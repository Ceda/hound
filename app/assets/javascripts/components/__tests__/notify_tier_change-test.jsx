import React from 'react';
import renderer from 'react-test-renderer';

import NotifyTierChange from '../notify_tier_change.js';

it('renders plans appropriately', () => {
  const repo = {
    id: 1,
    name: "Test repo"
  }
  const plans = [
    {name: "Chihuahua", price: 49, allowance: 4, current: true},
    {name: "Labrador", price: 99, allowance: 10},
    {name: "Great Dane", price: 249, allowance: 30},
  ]

  const component = renderer.create(
    <NotifyTierChange
      authenticity_token = "csrf_token"
      plans = {plans}
      repo_id = {repo.id}
      repo_name = {repo.name}
    />

  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
