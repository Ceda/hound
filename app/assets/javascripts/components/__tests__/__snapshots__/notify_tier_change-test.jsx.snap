exports[`test renders plans appropriately 1`] = `
<section
  className="tier-change-container">
  <aside
    className="tier-change-plans">
    <h3>
      Plans
    </h3>
    <div
      className="plan plan-vertical current">
      <div
        className="marker-wrapper">
        <span
          className="current-plan">
          Current Plan
        </span>
      </div>
      <div
        className="plan-divider">
        <h5
          className="plan-title">
          Chihuahua
        </h5>
        <div
          className="plan-allowance">
          Up to 
          <strong>
            4
          </strong>
           Repos
        </div>
      </div>
      <div
        className="plan-price">
        $
        49
         
        <small>
          month
        </small>
      </div>
    </div>
    <div
      className="plan plan-vertical new">
      <div
        className="marker-wrapper">
        <span
          className="new-plan">
          New Plan
        </span>
      </div>
      <div
        className="plan-divider">
        <h5
          className="plan-title">
          Labrador
        </h5>
        <div
          className="plan-allowance">
          Up to 
          <strong>
            10
          </strong>
           Repos
        </div>
      </div>
      <div
        className="plan-price">
        $
        99
         
        <small>
          month
        </small>
      </div>
    </div>
    <div
      className="plan plan-vertical">
      <div
        className="plan-divider">
        <h5
          className="plan-title">
          Great Dane
        </h5>
        <div
          className="plan-allowance">
          Up to 
          <strong>
            30
          </strong>
           Repos
        </div>
      </div>
      <div
        className="plan-price">
        $
        249
         
        <small>
          month
        </small>
      </div>
    </div>
  </aside>
  <div
    className="tier-change-content">
    <h1>
      Pricing: Change of Plans
    </h1>
    <section
      className="tier-change-description">
      <div
        className="allowance large">
        Private Repos
        <strong>
          4
          /
          4
        </strong>
      </div>
      <p>
        <strong>
          Activating \"
          Test repo
          \" will change the price you pay per month.
        </strong>
      </p>
      <p>
        You\'ll be charged an extra $
        50
         a month ($
        99
         total).
      </p>
      <p>
        Upgrade to continue your activation of \"
        Test repo
        \"
      </p>
    </section>
    <a
      className="repo-toggle tier-change-accept"
      href="javascript:void(0);"
      onClick={[Function]}>
      Upgrade
    </a>
    <a
      className="button tier-change-cancel"
      href="/repos">
      Cancel
    </a>
  </div>
</section>
`;
